### ISSUE TITLE
Figure out a way for GKE

### ISSUE THREAD

**Issue by SantoDE** [containous/maesh#208]
#### Proposal

There are multiple options I like to discuss:

- Can we make a specific installation guide by installing Core-DNS on GKE maybe? If possible?
- Support kube-dns over Core-DNS?
- Other options?

 #### Background

At the moment, GKE does nut support running Core-DNS. As Core-DNS is a requirement for Maesh in order to run, we're currently not supporting GKE. We want to change that.

**Comment by angelbarrera92** [comment]
+1 here

**Comment by SantoDE** [comment]
Would be fixed by #246

**Comment by dtomcej** [comment]
@SantoDE @mmatur can we confirm that 246 fixed this?

**Comment by donnyv12** [comment]
@dtomcej I just tried with 246 merged and still get an error. I'm using GKE 1.14 if that matters: 

```
$ helm install --name=maesh --namespace=maesh maesh/maesh --set kubedns=true
Error: Job failed: BackoffLimitExceeded
$ k logs -f maesh-prepare-dlqw2
time="2019-09-24T15:02:49Z" level=info msg="Checking Cluster..."
2019/09/24 15:02:49 command prepare error: error during cluster check: coredns does not exist in namespace kube-system
```

**Comment by donnyv12** [comment]
From looking at the tiller logs, I see: 

```
[tiller] 2019/09/24 15:06:58 preparing install for maesh
[storage] 2019/09/24 15:06:58 getting release history for "maesh"
[tiller] 2019/09/24 15:06:58 rendering maesh chart using values
[tiller] 2019/09/24 15:06:58 performing install for maesh
[tiller] 2019/09/24 15:06:58 executing 4 pre-install hooks for maesh
[tiller] 2019/09/24 15:06:58 deleting pre-install hook maesh-prepare for release maesh due to "before-hook-creation" policy
[kube] 2019/09/24 15:06:58 Starting delete for "maesh-prepare" ServiceAccount
[kube] 2019/09/24 15:06:58 building resources from manifest
[kube] 2019/09/24 15:06:58 creating 1 resource(s)
[kube] 2019/09/24 15:06:58 warning: failed to load schema: SchemaError(io.k8s.api.extensions.v1beta1.RunAsGroupStrategyOptions): invalid object doesn't have additional properties
[kube] 2019/09/24 15:06:58 Watching for changes to ServiceAccount maesh-prepare with timeout of 5m0s
[kube] 2019/09/24 15:06:58 Add/Modify event for maesh-prepare: ADDED
[tiller] 2019/09/24 15:06:58 deleting pre-install hook maesh-prepare-role for release maesh due to "before-hook-creation" policy
[kube] 2019/09/24 15:06:58 Starting delete for "maesh-prepare-role" ClusterRole
[kube] 2019/09/24 15:06:58 building resources from manifest
[kube] 2019/09/24 15:06:58 creating 1 resource(s)
[kube] 2019/09/24 15:06:58 warning: failed to load schema: SchemaError(io.k8s.api.extensions.v1beta1.RunAsGroupStrategyOptions): invalid object doesn't have additional properties
[kube] 2019/09/24 15:06:58 Watching for changes to ClusterRole maesh-prepare-role with timeout of 5m0s
[kube] 2019/09/24 15:06:58 Add/Modify event for maesh-prepare-role: ADDED
[tiller] 2019/09/24 15:06:58 deleting pre-install hook maesh-prepare for release maesh due to "before-hook-creation" policy
[kube] 2019/09/24 15:06:58 Starting delete for "maesh-prepare" ClusterRoleBinding
[kube] 2019/09/24 15:06:58 building resources from manifest
[kube] 2019/09/24 15:06:58 creating 1 resource(s)
[kube] 2019/09/24 15:06:58 warning: failed to load schema: SchemaError(io.k8s.api.extensions.v1beta1.RunAsGroupStrategyOptions): invalid object doesn't have additional properties
[kube] 2019/09/24 15:06:58 Watching for changes to ClusterRoleBinding maesh-prepare with timeout of 5m0s
[kube] 2019/09/24 15:06:58 Add/Modify event for maesh-prepare: ADDED
[tiller] 2019/09/24 15:06:58 deleting pre-install hook maesh-prepare for release maesh due to "before-hook-creation" policy
[kube] 2019/09/24 15:06:58 Starting delete for "maesh-prepare" Job
[kube] 2019/09/24 15:06:58 Using reaper for deleting "maesh-prepare"
[kube] 2019/09/24 15:06:58 building resources from manifest
[kube] 2019/09/24 15:06:58 creating 1 resource(s)
[kube] 2019/09/24 15:06:58 warning: failed to load schema: SchemaError(io.k8s.api.extensions.v1beta1.RunAsGroupStrategyOptions): invalid object doesn't have additional properties
[kube] 2019/09/24 15:06:58 Watching for changes to Job maesh-prepare with timeout of 5m0s
[kube] 2019/09/24 15:06:58 Add/Modify event for maesh-prepare: ADDED
[kube] 2019/09/24 15:06:58 maesh-prepare: Jobs active: 0, jobs failed: 0, jobs succeeded: 0
[kube] 2019/09/24 15:06:58 Add/Modify event for maesh-prepare: MODIFIED
[kube] 2019/09/24 15:06:58 maesh-prepare: Jobs active: 1, jobs failed: 0, jobs succeeded: 0
[kube] 2019/09/24 15:07:00 Add/Modify event for maesh-prepare: MODIFIED
[tiller] 2019/09/24 15:07:00 warning: Release maesh pre-install maesh/templates/hooks/prepare-hook.yaml could not complete: Job failed: BackoffLimitExceeded
[tiller] 2019/09/24 15:07:00 failed install perform step: Job failed: BackoffLimitExceeded
```

Emphasis added by me,

**Comment by dtomcej** [comment]
@donnyv12, the code was merged, but it is not in a public release yet. You would have to use a custom built image to test.

**Comment by SantoDE** [comment]
@dtomcej yes we can confirm. When we developed that, I tested it on GKE with a private image. @donnyv12 would you be willing to test? I could provide you with a private image to test.

**Comment by donnyv12** [comment]
I would be happy to test it.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by xavierbaude** [comment]
It's working from my side without custom image, since the PR is merged. :)

**Comment by mmatur** [comment]
Fixed by #246
